rfcs/text/0011_reporting_as_an_api.md

```diff
@@ -0,0 +1,284 @@
+- Start Date: 2020-04-23
+- RFC PR: (leave this empty)
+- Kibana Issue: (leave this empty)
+
+# Summary
+
+The reporting plugin is migrating to a purely REST API interface, deprecating page-level integrations such as Dashboard and Discover.
+
+# Basic example
+
+Currently, reporting does expose an API for Dashboard exports as seen below.
+
+```sh
+# Massively truncated URL
+curl -x POST http://localhost:5601/api/reporting/generate/printablePdf?jobParams=%28browserTimezone%3AAmerica%2FLos_Angeles%2Clayout...
+```
+
+Going forth, reporting would only offer a JSON-based REST API, deprecating older ad-hoc solutions:
+
+```sh
+curl -x POST http://localhost:5601/api/reporting/pdf
+{
+  “baseUrl”: “/my/kibana/page/route?foo=bar&reporting=true”,
+  "waitUntil": {
+    "event": “complete”,
+  },
+  “viewport”: {
+    “width”: 1920,
+    “height”: 1080,
+    "scale": 1
+  },
+  "mediaType": "screen"
+}
+```
+
+A simple JSON response is returned, with an identifier to query for status.
+
+```json
+{
+  "id": "123"
+}
+```
+
+Further information can be found via GET call with the job's ID:
+
+```sh
+curl http://localhost:5601/api/reporting/123/status
+```
+
+# Motivation
+
+The reporting functionality that currently exists in Kibana was originally purpose-built for the Discover, Dashboard and Canvas applications. Because of this, reportings underlying technologies and infrastructure are hard to improve upon and make generally available for pages across Kibana. Currently, the team has to:
```

Comment:
```suggestion
The Reporting plugin that currently exists in Kibana was originally purpose-built for the Discover, Dashboard and Canvas applications. Because of this, Reporting's underlying technologies and infrastructure are hard to improve upon and make generally available for pages across Kibana. Currently, the team has to:
```